tests/Integration/Behaviour/Features/Context/CommonFeatureContext.php

```diff
@@ -195,6 +197,14 @@ public static function clearCacheAfterFeature()
         self::clearCache();
     }
 
+    /**
+     * @BeforeScenario @reset-context-before-feature
+     */
+    public static function resetContextBeforeFeature()
```

Comment:
I don't think this is necessary Since this is already handled in `ContextFeatureContext` before each scenario

But the `ContextFeatureContext` is not associated to every suite for now, only the cart one And you're dealing with the `order` suite Besides resetting context on each scenario makes sense for cart tests but not necessarily for Order

So I guess it can be useful to be able to enable this behaviour on demande thanks to a tag, however I think the current implementation misses the point compared to `ContextFeatureContext` In the original context the `ContextMocker` is initialized on `@BeforeScenario` and then reset on `@AfterScenario` Here you only setup the mocker but never reset it

So you would need to endpoints here that are related to the same tag:
- one `@BeforeScenario @reset-context-before-feature` which does the same thing as `ContextFeatureContext::mockContext`
- one `@AfterScenario @reset-context-before-feature` which resets it to avoid messing with the following tests

However you are setting the tag on the Feature level, but use a `@BeforeScenario` directive meaning the context will be reset on EACH scenario in the feature file, if you only want to mock and rest your context once per feature then you should use these instead:
- `@BeforeFeature @reset-context-before-feature`
- `@AfterFeature @reset-context-before-feature`

Reply:
Fixed.